homeassistant/components/fitbit/sensor.py

```diff
@@ -371,7 +371,7 @@ def __init__(
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return self._name
+        return "fitbit_" + self._name
```

Comment:
The 500 error is caused during setup. During setup, these sensors do not exist, thus this change should not be able to solve a 500 error.

What this change will do, is be breaking for existing users.

Reply:
Hey Frenck,

Sorry. I didn't think this PR had been finalised/submitted. I thought it would wait until I had time to do the tests and complete the checklist before it submitted.

I completely agree on both counts. This code is not to fix the 500 error, as mentioned in the opening comment. There is a custom component that has a workaround for the 500 error. That is how I got the Fitbit sensors working on my system. However, it makes much more sense to me to have `sensor.fitbit_resting_heart_rate` than `sensor.resting_heart_rate`.

I agree it is a breaking change for existing users, but I know that I would rather have the sensor readings listed together, rather than a few dozen sensors distributed throughout the sensor domain. As it is, there is no way to filter the Fitbit entities using the dev tools or using filter cards etc (without listing them individually). As I can't find a device or integration that they belong to either. Now, I'm either missing something and using it incorrectly, or there are a few ways to improve things.

Either way, as I mentioned, I am willing to learn, and do certain tasks. It may just me longer than usual as I'm both disabled and busy. If the custom component has done something that removes the device or integration info, that would normally be available in this repo, then please accept my apologies.

Thanks again, Frenck
John

Reply:
> However, it makes much more sense to me to have sensor.fitbit_resting_heart_rate than sensor.resting_heart_rate.

Agreed, but we should implement unique IDs and configuration flows instead. Changing this in the current state, is a breaking change for existing users which I won't approve, as there is no direct need to (its just opinion based at this point).